src/libraries/System.Security.Cryptography.Encoding/tests/PemEncodingTests.cs

```diff
@@ -0,0 +1,548 @@
+            Assert.Equal("!PANIC~~~", content[fields.Label]);
+        }
+
+        [Fact]
+        public static void TryFind_True_MultiPem()
+        {
+            string content = @"
+-----BEGIN EC PARAMETERS-----
+BgUrgQQACg==
+-----END EC PARAMETERS-----
+-----BEGIN EC PRIVATE KEY-----
+MHQCAQEEIIpP2qP/mGWDAojQDNrNfUHwYGNPKeO6VLt+POJeCJ3OoAcGBSuBBAAK
+oUQDQgAEeDThNbdvTkptgvfNOpETlKBcWDUKs9IcQ/RaFeBntqt+6J875A79YhmD
+D7ofwIDcVqzOJQDhSN54EQ17CFQiwg==
+-----END EC PRIVATE KEY-----
+";
+            ReadOnlySpan<char> pem = content;
+            List<string> labels = new List<string>();
+            while (PemEncoding.TryFind(pem, out PemFields fields))
+            {
+                labels.Add(pem[fields.Label].ToString());
+                pem = pem[fields.Location.End..];
+            }
+
+            Assert.Equal(new string[] { "EC PARAMETERS", "EC PRIVATE KEY" }, labels);
+        }
+
+        [Fact]
+        public static void TryFind_True_FindsPemAfterPemWithInvalidBase64()
+        {
+            string content = @"
+-----BEGIN TEST-----
+$$$$
+-----END TEST-----
+-----BEGIN TEST2-----
+Zm9v
+-----END TEST2-----";
+            Assert.True(PemEncoding.TryFind(content, out PemFields fields));
+            Assert.Equal("TEST2", content[fields.Label]);
+            Assert.Equal("Zm9v", content[fields.Base64Data]);
+        }
+
+        [Fact]
+        public static void TryFind_True_FindsPemAfterPemWithInvalidLabel()
+        {
+            string content = @"
+-----BEGIN ------
+YmFy
+-----END ------
+-----BEGIN TEST2-----
+Zm9v
+-----END TEST2-----";
+            Assert.True(PemEncoding.TryFind(content, out PemFields fields));
+            Assert.Equal("TEST2", content[fields.Label]);
+            Assert.Equal("Zm9v", content[fields.Base64Data]);
+        }
+
+        [Fact]
+        public static void TryFind_False_Empty()
+        {
+            Assert.False(PemEncoding.TryFind(string.Empty, out _));
+        }
+
+        [Fact]
+        public static void TryFind_False_PostEbBeforePreEb()
+        {
+            string content = "-----END TEST-----\n-----BEGIN TEST-----\nZm9v";
+            Assert.False(PemEncoding.TryFind(content, out _));
+        }
+
+        [Theory]
+        [InlineData("\tOOPS")]
+        [InlineData(" OOPS")]
+        [InlineData("-OOPS")]
+        [InlineData("te\x7fst")]
+        [InlineData("te\x19st")]
+        public static void TryFind_False_InvalidLabel(string label)
+        {
+            string content = $"-----BEGIN {label}-----\nZm9v\n-----END {label}-----";
+            Assert.False(PemEncoding.TryFind(content, out _));
+        }
+
+        [Fact]
+        public static void TryFind_False_InvalidBase64()
+        {
+            string content = "-----BEGIN TEST-----\n$$$$\n-----END TEST-----";
+            Assert.False(PemEncoding.TryFind(content, out _));
+        }
+
+        [Fact]
+        public static void TryFind_False_PrecedingLinesAndSignificantCharsBeforePreeb()
```

Comment:
I guess even under full lax, the character before the starting - of the pre-encapsulation boundary must not exist or be whitespace.